src/test/java/com/aol/cyclops2/functions/collections/extensions/standard/anyM/ReactiveStreamXTest.java

```diff
@@ -152,13 +150,31 @@ public void whenNilOrNot(){
     }
     @Test
     public void whenNilOrNotJoinWithFirstElement(){
-        
-        
+
+
         String res= of(1,2,3).visit((x,xs)-> xs.join(x>2? "hello" : "world"),()->"EMPTY");
         assertThat(res,equalTo("2world3"));
     }
-	
-	
 
+    public static class BooleanProxy {
```

Comment:
This is fine, there is AtomicBoolean in the JDK which might do and also MutableBoolean in cyclops-react which is the same thing, but as it is test code, I think this is fine.